solc/CommandLineInterface.cpp

```diff
@@ -947,7 +947,7 @@ bool CommandLineInterface::processInput()
 			formatter->printErrorInformation(*error);
 		}
 
-		if (!successful)
+		if (!successful && !errorRecovery)
```

Comment:
If we do this, then actOnInput should be changed - only output binaries if there was no error.

Reply:
Done in bf0828057a4545cfaa6325849ee1bab14296d111 isolated as a separate commit so this can be reviewed in isolation.